src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private final double firstLeg;
+    private final double secondLeg;
+    private final ColorSupplier color;
+
+    public Rectangle(ColorSupplier color, double firstLeg, double secondLeg) {
+        this.color = color;
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+    }
+
+    @Override
+    public double calculateArea() {
+        return firstLeg * secondLeg;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: rectangle, area: "
+                + calculateArea() + "sq. units, firstLeg: "
+                + firstLeg + " units, " + "secondLeg: "
+                + secondLeg + " units, "
+                + "color: " + color);
```

Comment:
Use `color.name()` to get the String representation of the enum constant instead of directly printing the `color` object. This ensures that you are not relying on the `toString()` method, which can be overridden and thus may not return the enum constant's name.